## Clustal

### Description

ClustalO and ClustalW are multiple sequence alignment programs for nucleotide and protein sequences. Clustal uses a method called pairwise progressive sequence alignment. This heuristic method first does a pairwise sequences alignment for all the sequence pairs that can be constructed from the sequence set. A dendrogram (guide tree) of the sequences is then done according to the pairwise similarity of the sequences. Finally a multiple sequence alignment is is constructed by aligning sequences in the order, defined by the guide tree.

* * *

### Available

##### Version on CSC's Servers

Taito: Clustal Omega 1.2.1  
Taito: ClustalW 2.1  
Taito: ClustalX 2.1

* * *

### Usage

In Taito, you can start ClustalO with command:

    clustalo

sample command:

    clustalo -i my-in-seqs.fa -o my-out-seqs.fa -v

ClustalW is starterd with command:

    clustalw

In NoMachine remote desktop you can start clustalx with command:

    clustalx2

* * *

### Discipline

Biosciences  

* * *

### References

* * *

### Support

* * *

### Manual

* * *